library/alloc/tests/vec.rs

```diff
@@ -1110,23 +1110,31 @@ fn test_from_iter_specialization_panic_during_iteration_drops() {
 }
 
 #[test]
-fn test_from_iter_specialization_panic_during_drop_leaks() {
-    static mut DROP_COUNTER: usize = 0;
+fn test_from_iter_specialization_panic_during_drop_doesnt_leak() {
+    static mut DROP_COUNTER_SHOULD_BE_DROPPED: usize = 0;
+    static mut DROP_COUNTER_DROPPED_TWICE: usize = 0;
```

Comment:
the name is quite confusing. nothing should be dropped twice

Reply:
I copied the variant name, but that also had a weird name.